flow/Trace.cpp

```diff
@@ -932,11 +978,14 @@ TraceEvent& TraceEvent::backtrace(const std::string& prefix) {
 	return detail(prefix + "Backtrace", platform::get_backtrace());
 }
 
-void TraceEvent::log() {
+void TraceEvent::log(bool useCurrentTime) {
 	if(!logged) {
 		init();
 		try {
 			if (enabled) {
+				if (useCurrentTime) {
```

Comment:
I made this a parameter initially, but I'm considering changing the behavior unconditionally to choose the time when `log` is called. Otherwise, if the time is different than when the event is initialized, the times of events may not be in order.